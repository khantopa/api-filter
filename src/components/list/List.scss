@import '../../styles/main.scss';

.container {
  width: 100%;
  padding: 40px 5%;
  .filter {
    @include grid(3);
    grid-gap: 30px;
    margin-bottom: 20px;
  }
  .table {
   @include card(100%, auto);
    box-sizing: border-box;
    padding: 20px;
    box-shadow: 0 5px 10px 5px rgba(0, 0, 0, .15);
    &_head {
      width: 100%;
      background: $dark;
      color: $light;
      height: 40px;
      font-weight: bold;
    }
    &_body {
      &_row {
        height: 40px;
        border-bottom: 1px solid $light;
        &:hover {
          background: $light;
        }
        a {
          text-decoration: none;
          color: $primary;
          font-weight: bold;
          &:hover {
            color: $dark;
          }
        }
      }
    }
  }
}

