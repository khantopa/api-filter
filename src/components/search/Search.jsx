import React from 'react';
import PropTypes from 'prop-types';
import './Search.scss';

const Search = (props) => {

  const handleChange = (e) => {
    props.onInputChange(e.target.value)
  }

  return (
    <div className="search">
      <label htmlFor={ props.label } className="search_label">{ props.label }:</label> 
      <input type="text" className="search_field" onChange={ handleChange } />
    </div>
  );
}

Search.propTypes = {
  onInputChange: PropTypes.func
}

export default Search;
