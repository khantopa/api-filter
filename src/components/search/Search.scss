@import '../../styles/main.scss';

.search {
  width: 100%;
  text-align: left;
  &_label {
    color: $dark;  
    margin-bottom: 40px;  
  }
  &_field {
    margin-top: 10px;
    @extend .form_field;
    text-align: left;
  }
}