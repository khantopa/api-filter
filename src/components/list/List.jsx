import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import axios from 'axios';
import './List.scss';

import Search from '../search/Search';

export default class List extends Component {
  constructor(props) {
    super(props);
    this.state = {
      lists: []
    }
  }

  

  onNameChange = async(name) => {
    axios.get(`http://165.22.60.22/api/v1/contacts?name=${name}`)
    .then(response => {
      if(response.data.data !== null) {
        this.setState({lists: response.data.data})
      } else {
        this.getAllData();
      }
    })
  }

  onCompanyChange = async(id) => {
    axios.get(`http://165.22.60.22/api/v1/contacts?company_id=${id}`)
    .then(response => {
      if(response.data.data !== null) {
        this.setState({lists: response.data.data})
      } else {
        this.getAllData();
      }
    })
  }

  onRevenueChange = async(value) => {
    axios.get(`http://165.22.60.22/api/v1/contacts?revenue_gte=${value}`)
    .then(response => {
      if(response.data.data !== null) {
        this.setState({lists: response.data.data})
      } else {
        this.getAllData();
      }
    })
  }

  componentDidMount = () => {
    this.getAllData();
  }

  getAllData = () => {
    axios.get('http://165.22.60.22/api/v1/contacts')
    .then(response => {
      this.setState({lists: response.data.data})
    })
    .catch(err => 'err');
  }

  render() {

    const tableRows = this.state.lists.map( row => (
      <tr key={row.id} className="table_body_row">
        <td className="table_body_row_col">
          <Link to={`/contacts/${row.id}`}>{row.name}</Link>
        </td>
        <td className="table_body_row_col">{ row.company.name }</td>
        <td className="table_body_row_col">{ row.company.revenue }</td>
      </tr>
    ));

    const table = (
      <table className="table">
        <thead className="table_head">
          <tr className="table_head_row">
            <td className="table_head_row_col">Name</td>
            <td className="table_head_row_col">Company</td>
            <td className="table_head_row_col">Revenue</td>
          </tr>
        </thead>
        <tbody className="table_body">
          { tableRows }
        </tbody>
      </table>
    );


    return (
      <div className="container">
        <h1>Lists</h1>
        <br/>
        <div className="filter">
          <Search label="Search by name" onInputChange={ this.onNameChange } />
          <Search label="Search by company id" onInputChange={ this.onCompanyChange }/>
          <Search label="Search by revenue" onInputChange={ this.onRevenueChange }/>
        </div>
        { table }
      </div>
    )
  }
}
