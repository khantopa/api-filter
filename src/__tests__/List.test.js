import List from '../components/list/List';
import Search from '../components/search/Search';


let wrapper;

beforeEach(() => {
  global.fetch = jest.fn();
  wrapper = shallow(<List /> );
});

afterEach(() => {
  wrapper.unmount();
});

it('should render List', () => {
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('should have 3 Search', () => {
  expect(wrapper.find(Search)).toHaveLength(3);
})




