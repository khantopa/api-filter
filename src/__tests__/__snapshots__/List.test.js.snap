// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render List 1`] = `
<div
  className="container"
>
  <h1>
    Lists
  </h1>
  <br />
  <div
    className="filter"
  >
    <Search
      label="Search by name"
      onInputChange={[Function]}
    />
    <Search
      label="Search by company id"
      onInputChange={[Function]}
    />
    <Search
      label="Search by revenue"
      onInputChange={[Function]}
    />
  </div>
  <table
    className="table"
  >
    <thead
      className="table_head"
    >
      <tr
        className="table_head_row"
      >
        <td
          className="table_head_row_col"
        >
          Name
        </td>
        <td
          className="table_head_row_col"
        >
          Company
        </td>
        <td
          className="table_head_row_col"
        >
          Revenue
        </td>
      </tr>
    </thead>
    <tbody
      className="table_body"
    />
  </table>
</div>
`;
